controller/linodemachine_controller_helpers.go

```diff
@@ -403,29 +402,13 @@ func getVPCInterfaceConfig(ctx context.Context, machineScope *scope.MachineScope
 		return nil, errors.New("vpc is not available")
 	}
 
-	var subnetID int
-	vpc, err := machineScope.LinodeClient.GetVPC(ctx, *linodeVPC.Spec.VPCID)
-	if err != nil {
-		logger.Error(err, "Failed to fetch LinodeVPC")
-
-		return nil, err
-	}
-	if vpc == nil {
-		logger.Error(nil, "Failed to fetch VPC")
-
-		return nil, errors.New("failed to fetch VPC")
-	}
-	if len(vpc.Subnets) == 0 {
+	if len(linodeVPC.Spec.Subnets) == 0 {
 		logger.Error(nil, "Failed to find subnet")
 
 		return nil, errors.New("failed to find subnet")
 	}
-	// Place node into the least busy subnet
-	sort.Slice(vpc.Subnets, func(i, j int) bool {
-		return len(vpc.Subnets[i].Linodes) > len(vpc.Subnets[j].Linodes)
-	})
```

Comment:
question: because we are picking a subnet in linodevpc controller now and saving it in the spec we no longer need to look for least busy subnet? Just want to clarify!

Reply:
Yeah, we never used this logic and are not sure if we will ever need it in future. Currently, all our VPCs have single subnet and we use the first subnet returned by GetVPC call [here](https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodemachine_controller_helpers.go#L428). We don't want to make linode API GetVPC call for each instance creation as it slows things down and also causes ratelimiting issues if too many machines are brought up at the same time. Hence, we are getting rid of this call and simplifying our logic to use the one subnet we have.